android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboard.java

```diff
@@ -676,18 +676,30 @@ private void sendKMWError(int lineNumber, String sourceId, String message) {
     }
   }
 
-  // Extract Unicode numbers (\\uxxxx) from a layer to character string.
-  // Ignores empty strings and layer names
-  // Returns: String
+  /**
+   * Extract Unicode numbers (\\u_xxxx_yyyy) from a layer to character string.
+   * Ignores empty strings and layer names
+   * Refer to web/source/osk/oskKey.ts
+   * @param ktext Unicode string in the format \\uxxxx_yyyy
+   * @return String to display on subkey
+   */
   protected String convertKeyText(String ktext) {
     String title = "";
     String[] values = ktext.split("\\\\u");
     int length = values.length;
     for (int j = 0; j < length; j++) {
       if (!values[j].isEmpty() && !values[j].contains("-")) {
-        int c = Integer.parseInt(values[j], 16);
-        // TODO: \\uxxxxxx will need to be handled with title.codePointAt(c)
-        title += String.valueOf((char) c);
+        // Split U_xxxx_yyyy
+        String[] codePoints = values[j].split("_");
+        for (String codePoint : codePoints) {
+          int codePointValue = Integer.parseInt(codePoint, 16);
+          if (((0x0 <= codePointValue) && (codePointValue <= 0x1F)) || ((0x80 <= codePointValue) && (codePointValue <= 0x9F))) {
+            continue;
+          } else {
+            // TODO: \\uxxxxxx will need to be handled with title.codePointAt(c)
```

Comment:
What is this TODO for?

Reply:
I added that in #357 but forget why. I'll remove it.